### PR TITLE
disable add to until page load

### PR DESCRIPTION
**Pull request by sunilit42**
### Description (*)
Open a configurable product having DropDown options
When the page is being loaded, you will notice that the dropdowns has no values
Click Add To Cart button during this time. The page will be reloaded and a notice message "Open a configurable product having DropDown options" will be shown

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=20964) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=20964) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: sunilaccorin<br/>:x: Sunil Patel<hr/>**Sunil Patel** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

**Comment by magento-engcom-team** [comment]
Hi @sunilit42. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Catalog/view/frontend/web/js/validate-product.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ define([
                     return false;
                 }
             });
+            $('.action.tocart').attr('disabled',false);
```
Comment (orlangur) [inline app/code/Magento/Catalog/view/frontend/web/js/validate-product.js]: Such selector is not acceptable, see https://devdocs.magento.com/guides/v2.3/coding-standards/code-standard-demarcation.html#business-logic-and-javascript

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/view/frontend/web/js/validate-product.js (hunk 1 above)
Comment (sunilit42) [reply app/code/Magento/Catalog/view/frontend/web/js/validate-product.js]: ok i am working on that

Comment (sunilit42) [reply app/code/Magento/Catalog/view/frontend/web/js/validate-product.js]: @orlangur 

fixed it, please check it.

**Comment by sunilit42** [comment]
@orlangur  i did changes, please check it again

**Comment by sidolov** [comment]
Hi @sunilit42  , looks like you made some commits with email different than in your GitHub profile, please, add email from commits to your profile!
Thank you!

File: app/code/Magento/Catalog/view/frontend/web/js/validate-product.js [hunk 2]
```diff
@@ -13,7 +13,8 @@ define([
     $.widget('mage.productValidate', {
         options: {
             bindSubmit: false,
-            radioCheckboxClosest: '.nested'
+            radioCheckboxClosest: '.nested',
+            addToCartButtonSelector: '.action.tocart'
```
Comment (orlangur) [inline app/code/Magento/Catalog/view/frontend/web/js/validate-product.js:17]: Again, CSS classes cannot be used as a JavaScript logic selectors. Please check examples in link I posted.

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/view/frontend/web/js/validate-product.js (hunk 2 above)
Comment (sunilit42) [reply app/code/Magento/Catalog/view/frontend/web/js/validate-product.js:17]: @orlangur  magento itself using that same class into add to cart js so i am using that
https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Catalog/view/frontend/web/js/catalog-add-to-cart.js#L23

let me know if still you want to change it

**Comment by sunilit42** [comment]
@sidolov 
what should i need to do it? 
can i create new one and close to this one?

**Comment by VladimirZaets** [comment]
Hi @sunilit42. You have two options. 
The first one: add email to Github that is set in git-config (git config --global user.email for look git email)
The second one: set your Github email to git config (use the same command with passing email address as the last parameter) and create new one PR

**Comment by sunilit42** [comment]
@VladimirZaets 

i created new pull request 
https://github.com/magento/magento2/pull/21007

**Pull request closed without merging**